### PR TITLE
test: validate inputs to acn components

### PR DESCRIPTION
**Pull request by QxBytes**
**Reason for Change**:

Validates some fields in requests to the cns, and also validates some fields in the cni args and network config. 

**This PR modifies the behavior of CNS:**
- If the primary interface ip is not a valid ip or cidr, an error will now be raised
- Same if the gateway ip address is not an ip address

**This PR modifies the behavior of CNI:**
- The container id, ifname, and bridge name passed into cni must fully consists of zero or more alphanumeric, dots, dashes, parentheses, or underscores. Otherwise, an error will be raised

**Issue Fixed**:

**Requirements**:

- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [X] adds unit tests
- [X] relevant PR labels added

**Notes**:
Please confirm there are no existing scenarios in cns where we have a primary interface name or gateway ip field that aren't ips.

### CODE REVIEW

File: cni/network/network.go [hunk 1]
```diff
@@ -1451,3 +1468,21 @@ func convertCniResultToInterfaceInfo(result *cniTypesCurr.Result) network.Interf
...
+		return errors.New("invalid network config value")
+	}
+
+	return nil
+}
+
+// returns true if the string fully consists of zero or more alphanumeric, dots, dashes, parentheses, or underscores
+func isValidString(value string) bool {
+	pattern := `^[a-zA-Z0-9._\-\(\) ]*$`
+	re := regexp.MustCompile(pattern)
```
Comment (rbtr) [inline cni/network/network.go:1486]: It's possible the Go compiler is smart enough to optimize this away, but I doubt it - you should move this to a package var so that it is not being recompiled on every request. The pattern could also be inlined instead of creating an extra intermediate var.

File: cns/NetworkContainerContract.go [hunk 2]
```diff
@@ -131,9 +132,22 @@ func (req *CreateNetworkContainerRequest) Validate() error {
...
+		return errors.Wrapf(ErrInvalidIP, "GatewayIPAddress %s is not a valid ip address", req.IPConfiguration.GatewayIPAddress)
+	}
 	return nil
 }
 
+func isValidIP(ipStr string) bool {
+	if ip, _, err := net.ParseCIDR(ipStr); err == nil {
+		return ip != nil
+	}
+	return net.ParseIP(ipStr) != nil
```
Comment (rbtr) [inline cns/NetworkContainerContract.go:148]: This block is a little hard to follow (because it's trying to be so compact). I think readability should be preferred here, where I need to be able to quickly grok what the valid/invalid inputs are and don't necessarily remember what `net.ParseIP` returns - is it an `error`? Is it a `net.IP` or a `*net.IP`? The answer significantly changes my interpretation of this function and its correctness.

**Review by rbtr — CHANGES_REQUESTED** [review]

File: cni/network/network.go [hunk 3]
```diff
@@ -1468,3 +1491,19 @@ func convertCniResultToInterfaceInfo(result *cniTypesCurr.Result) network.Interf
...
+		return errors.New("invalid network config value")
+	}
+
+	return nil
+}
+
+// returns true if the string fully consists of zero or more alphanumeric, dots, dashes, parentheses, or underscores
+func isValidString(value string) bool {
+	return validRegexp.MatchString(value)
+}
```
Comment (rbtr) [inline cni/network/network.go:1509]: move the doc to the regexp declaration, rename it to something like `allowedInput` and drop this extra function def entirely

**Review by rbtr — APPROVED** [review]

**Review by paulyufan2 — APPROVED** [review]

**Comment by QxBytes** [comment]
Wanted to see if I could add more cni input checks-- hold off on merging for now

**Comment by QxBytes** [comment]
confirmed certain sections of code are unused and so ready to merge
